### PR TITLE
Add Google API support

### PR DESCRIPTION
**Pull request by solocommand**
This PR adds a required environment variable for `GOOGLE_APPLICATION_CREDENTIALS`. On dev, this will attempt to default to `./google-cloud.json`.

Contact if you need the credentials -- if you are not testing GC APIs you can just create an empty JSON file in that location.

In production, the environment variable and secret file distribution has already been configured, so this is good to merge once reviewed.

This can be utilized by something like the following:
```js

const { google, auth } = require('../connections/google');

const runSample = async () => {
  const api = google.analyticsreporting({
    version: 'v4',
    auth: await auth(['https://www.googleapis.com/auth/analytics']),
  });

  const dateRanges = [
    { startDate: '2018-03-17', endDate: '2018-03-24' },
    { startDate: '14daysAgo', endDate: '7daysAgo' },
  ];

  const res = await api.reports.batchGet({
    requestBody: {
      reportRequests: [
        {
          viewId: '31090033',
          dateRanges,
          metrics: [
            { expression: 'ga:users' },
          ],
        },
      ],
    },
  });
  console.log(res.data); // eslint-disable-line no-console
  return res.data;
};

runSample();
```

### CODE REVIEW

File: src/connections/google/auth.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const { client_email, private_key } = require('../../env').GOOGLE_APPLICATION_CREDENTIALS; // eslint-disable-line camelcase
+
+/**
+ *
+ * @param {Array} endpoints Google API endpoints to be authorized for this client
+ * @returns JWT
+ */
+module.exports = async (endpoints = []) => {
+  const client = new JWT(client_email, null, private_key, endpoints);
+  await client.authorize((err) => {
```
Comment (zarathustra323) [inline src/connections/google/auth.js]: I don't think this will actually await, as the client.authorize call will immediately execute, and the Promise.resolve function will not be immediately returned. Instead, you should wrap the authorize call with a promise beforehand, and then await:

```js
const authorize = new Promise((resolve, reject) => {
  client.authorize((err) => {
    if (err) {
      reject(err);
    } else {
      resolve();
    }
  })
}); 
await authorize();
```

File: src/env.js [hunk 2]
```diff
@@ -30,6 +30,14 @@ const nonemptystr = makeValidator((v) => {
   return trimmed;
 });
 
+const jsonFile = makeValidator((v) => {
```
Comment (zarathustra323) [inline src/env.js]: make this lowercase to match the other env functions: `jsonfile`

File: src/env.js [hunk 3]
```diff
@@ -30,6 +30,14 @@ const nonemptystr = makeValidator((v) => {
...
+  if (v === undefined || v === null || v === '') {
+    throw new Error('Expected a non-empty string');
+  }
+  // eslint-disable-next-line import/no-dynamic-require
+  return require(v); // eslint-disable-line global-require
```
Comment (zarathustra323) [inline src/env.js]: I don't think the validator should do the require, should it?
It should only be concerned with validating that it is a file and it exists

Comment (solocommand) [reply src/env.js]: I did this intentionally, as requiring will also parse the file as js/json. Either having a missing file or a file with invalid json will cause an unhandled rejection from the google api client, so I thought it prudent to catch that up front.
